lib/metasploit/framework/login_scanner/teamcity.rb

```diff
@@ -134,9 +138,26 @@ class TeamCityError < StandardError; end
         class StackLevelTooDeepError < TeamCityError; end
         class NoPublicKeyError < TeamCityError; end
         class PublicKeyExpiredError < TeamCityError; end
-        class DecryptionException < TeamCityError; end
+        class DecryptionError < TeamCityError; end
         class ServerNeedsSetupError < TeamCityError; end
 
+        # Checks if the target is JetBrains TeamCity. The login module should call this.
+        #
+        # @return [Boolean] TrueClass if target is TeamCity, otherwise FalseClass
+        def check_setup
```

Comment:
Not a blocker; we should probably wire up `check_setup` to framework itself at some point - as this doesn't get invoked otherwise